src/TorchSharp/TorchVision/dsets/CIFAR.cs

```diff
@@ -0,0 +1,316 @@
+        /// <param name="train">If true, creates dataset from training set, otherwise creates from test set</param>
+        /// <param name="download">If true, downloads the dataset from the internet and puts it in root directory. If dataset is already downloaded, it is not downloaded again</param>
+        /// <param name="target_transform">A transform that takes in the target and transforms it.</param>
+        /// <returns>A CIFAR dataset.</returns>
+        public static Dataset CIFAR100(string root, bool train, bool download = false, ITransform target_transform = null)
+        {
+            return new Modules.CIFAR100(root, train, "https://www.cs.toronto.edu/~kriz/", download, target_transform);
+        }
+
+        /// <summary>
+        /// CIFAR100 Dataset.
+        /// </summary>
+        /// <remarks>
+        /// The dataset for this example can be found at: https://www.cs.toronto.edu/~kriz/cifar.html
+        /// </remarks>
+        /// <param name="root">Root directory of dataset where directory CIFAR{10,100} exists or will be saved to if download is set to true.</param>
+        /// <param name="train">If true, creates dataset from training set, otherwise creates from test set</param>
+        /// <param name="download">If true, downloads the dataset from the internet and puts it in root directory. If dataset is already downloaded, it is not downloaded again</param>
+        /// <param name="target_transform">A function that takes in the target and transforms it.</param>
+        /// <returns>A CIFAR dataset.</returns>
+        public static Dataset CIFAR100(string root, bool train, bool download, Func<Tensor, Tensor> target_transform)
+        {
+            return new Modules.CIFAR100(root, train, "https://www.cs.toronto.edu/~kriz/", download, new torchvision.Lambda(target_transform));
+        }
+    }
+
+    namespace Modules
+    {
+        internal abstract class CIFAR : Dataset
+        {
+            protected void Download(string targetDir, string baseUrl, string dataset, string fileName)
+            {
+
+                if (!Directory.Exists(targetDir)) {
+                    Directory.CreateDirectory(targetDir);
+                }
+
+                DownloadFile(fileName, targetDir, baseUrl);
+
+                if (!File.Exists(Path.Combine(targetDir, $"{dataset}-binary.bin"))) {
+                    DecompressFile(fileName, targetDir, targetDir);
+                }
+            }
+            private static void DecompressFile(string file, string sourceDir, string targetDir)
+            {
+                Utils.Decompress.ExtractTGZ(Path.Combine(sourceDir, file), targetDir);
+            }
+
+            private void DownloadFile(string file, string target, string baseUrl)
+            {
+#if NETSTANDARD2_0_OR_GREATER
+                var filePath = NSPath.Join(target, file);
+#else
+                var filePath = Path.Join(target, file);
+#endif // NETSTANDARD2_0_OR_GREATER
```

Comment:
what you think moving these into helper method to reduce the number of '#if' inserted in the sources?